### DISCUSSION TITLE
Is there a builtin way for the JWT middleware to not early return when unauthenticated?

### DISCUSSION THREAD

**Discussion by itsramiel** [honojs/hono#3530] [Q&A]
Hi,

I have a route which accepts both auth/unauth requests. I would like to use the JWT middleware cause I use access Tokens but the JWT seems to early return if the user is unauthed but rather I would like to have an an optional jwtPayload then.

Is that in some way supported?

**Comment by maou-shonen** [comment]
Although it is not mentioned in the documentation, you can use it this way.

```ts
import { Hono } from "hono"
import { jwt, decode,verify } from "hono/jwt"
import { HTTPException } from "hono/http-exception"

new Hono().get(
  "/",
  async (c, next) => {
    const token = c.req.header("authorization")?.split(" ")[1]
    if (!token) {
      throw new HTTPException(403)
    }

    const { header, payload } = decode(token)

    // When you're ready, validate it.
    const verifiedPayload = await verify(token, "secret")

    return await next()
  },
  // Of course, you can also let the JWT middleware validate it.
  jwt({
    secret: "secret",
  }),
  (c) => c.text("hi"),
)
```

**Reply by itsramiel** [reply]
So basically writing my own middleware? Also in your example you are throwing if there is no token but my case is that it would not throw or early return but instead have no `jwtPayload`.

**Reply by maou-shonen** [reply]
This is just an example; you should write your own.
This line in the example should already be what you're looking for.
```ts
const { header, payload } = decode(token)
```

**Comment by DeityLamb** [comment]
You can use a [combine middleware](https://hono.dev/docs/middleware/builtin/combine)
```ts
import { except, some } from 'hono/combine';

app.use(some(jwt({ secret }), () => true));
```
